### ISSUE TITLE
LongHashSet can hang with 100% CPU usage during certain mutations

### ISSUE THREAD

**Issue by aewhite** [eclipse/eclipse-collections#551]
While working on a toy project I ran into a case where the `LongHashSet` would "freeze" after calling `add`. I managed to debug the issue to a while loop in the `probeThree` method but that was as far as I got. Here is a kotlin program that will reproduce it every time for me.

```kotlin
import org.eclipse.collections.api.set.primitive.MutableLongSet
import org.eclipse.collections.impl.factory.primitive.LongSets
import java.util.*


fun main(args: Array<String>) {
    val random = Random(-6819587229044058720)
    val expectedCardinality = random.nextInt(100000) + 100000

    val bitSmear = BitSmear2()
    (1..expectedCardinality).forEach {
        val hash = random.nextLong() and 0x7FFFFFFFFFFFFFFF
        bitSmear.offer(hash)
    }
}

private class BitSmear2(val bits: Int = 128, val maxUpperBound: Long = Long.MAX_VALUE) {
    var smearCount = 0
    var base = Math.pow(maxUpperBound.toDouble(), 1.0/bits)
    var threshold = (maxUpperBound/base).toLong()
    var upperBound = maxUpperBound
    val interestingHashes: MutableLongSet = LongSets.mutable.empty()

    fun offer(hash: Long) {
        if (hash > upperBound) { return }

        interestingHashes.add(hash)

        if (hash >= threshold) {
            smear()
        }
    }

    private fun smear() {
        while (interestingHashes.removeIf { it >= threshold }) {
            println(1.0/((upperBound.toDouble()-threshold)/maxUpperBound))
            smearCount++
            updateState()
        }
    }

    private fun updateState() {
        base = Math.pow(threshold.toDouble(), 1.0/bits)
        upperBound = threshold
        threshold = (threshold/base).toLong()
    }
}
```

This is with EC version 9.2.0 on JDK 1.8.0_171 on OSX.

**Comment by nikhilnanivadekar** [comment]
@mohrezaei @kristeno can you please take a look at this issue?

**Comment by nikhilnanivadekar** [comment]
Fixed via #552 . The fix will be available as a part of 10.0 Milestone release which will be released next Friday.
